File: halcyon-attachment/src/main/java/com/marshal/halcyon/attachment/controller/SysAttachmentCategoryController.java
package com.marshal.halcyon.attachment.controller;


import com.marshal.halcyon.core.controller.BaseController;
import com.marshal.halcyon.core.component.SessionContext;
import com.marshal.halcyon.core.util.RequestHelper;
import com.marshal.halcyon.core.entity.ResponseData;

import com.marshal.halcyon.attachment.entity.SysAttachmentCategory;
import com.marshal.halcyon.attachment.service.SysAttachmentCategoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

@RestController
public class SysAttachmentCategoryController extends BaseController {

    @Autowired
    private SysAttachmentCategoryService service;


    @RequestMapping(value = "/sys/attachment/category/query")
    public ResponseData query(SysAttachmentCategory dto,
                              @RequestParam int pageNum,
                              @RequestParam int pageSize,
                              HttpServletRequest request) {
        SessionContext sessionContext = RequestHelper.getSessionContext(request);
        return new ResponseData(service.select(dto, pageNum, pageSize));
    }

    @RequestMapping(value = "/sys/attachment/category/save")
    public ResponseData update(@RequestBody SysAttachmentCategory dto,
                               HttpServletRequest request) {
        if (getValidator().hasError(dto)) {
            return new ResponseData(false, getValidator().getErrors(dto));
        }
        service.save(dto);
        return new ResponseData();
    }

    @RequestMapping(value = "/sys/attachment/category/delete")
    public ResponseData delete(@RequestParam("selectedIds") Long[] selectedIds) {
        service.batchDelete(selectedIds);
        return new ResponseData(true, "删除成功");
    }
}

File: halcyon-web/src/main/resources/static/js/controller/attachment/sysAttachmentCategoryController.js
//控制层
app.controller("sysAttachmentCategoryController", function ($scope, $controller, sysAttachmentCategoryService, hrEmployeeService) {
    //继承
    $controller("baseController", {$scope: $scope});

    //初始化
    $scope.initForAdd = function () {
        $scope.sysAttachmentCategory = {};
        $scope.getEmpOptions();
    }
    $scope.initForUpdate = function (id) {
        sysAttachmentCategoryService.queryById(id).success(function (data) {
            $scope.sysAttachmentCategory = data;
        });
        $scope.getEmpOptions();
    }

    //crud
    //查询,初始化查询条件为空
    $scope.condition = {};
    $scope.query = function (pageNum, pageSize) {
        sysAttachmentCategoryService.query(pageNum, pageSize, $scope.condition).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //保存
    $scope.save = function () {
        sysAttachmentCategoryService.save($scope.sysAttachmentCategory).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }
    //删除
    $scope.delete = function () {
        $scope.deleteConfirm(deleteRows);
    }

    function deleteRows() {
        sysAttachmentCategoryService.delete($scope.selectedIds).success(function (responseData) {
            $scope.parseResponse(responseData);
        });
    }

});